tpm2_pytss/TCTI.py

```diff
@@ -0,0 +1,70 @@
+"""
+SPDX-License-Identifier: BSD-2
+"""
+
+from ._libtpm2_pytss import ffi
+
+from .utils import _chkrc
+
+
+class TCTI:
+    def __init__(self, ctx):
+        self._v1 = ffi.cast("TSS2_TCTI_CONTEXT_COMMON_V1 *", ctx)
+        if self._v1.version == 2:
+            self._v2 = ffi.cast("TSS2_TCTI_CONTEXT_COMMON_V2 *", ctx)
+        else:
+            self._v2 = None
+        self._ctx = ctx
+
+    @property
+    def tcti_context(self):
```

Comment:
I don't want public API that returns an FFI pointer, so I want this one to go away. Where you call it, just call ._ctx and consider it "Package Scope".

Reply:
Ah, I see, my general idea was to enable flexibility on how applications configure/use TCTI and by only expecting a certain property to be available it provides some room to do non-standard things.
I can always rename the property to _tcti_context instead, to make it clear that it's not expected to by used outside of the package

Reply:
@williamcroberts Maybe I understood this incorrectly, so please help me out here.

I think we **need** the TCTI class to expose the underlying `cdata` object. The FAPI returns only a TCTI object. If there was no way to get the `cdata` object from the TCTI class, a user couldn't get the `cdata` TCTI from FAPI.

Reply:
The class itself will need to make the context as the cdata pointer available for calls. However, in Python, anything not prefixed with an underscore is considered public API and free-to-call by end users. Thus, I don't want end users directly asking for the cdata pointer, so all I am asking for here is to change.

Reply:
@joholl The class itself will need to make the context as the cdata pointer available for calls. However, in Python, anything not prefixed with an underscore is considered public API and free-to-call by end users. Thus, I don't want end users directly asking for the cdata pointer, so all I am asking for here is to change.

Reply:
> Ah, I see, my general idea was to enable flexibility on how applications configure/use TCTI and by only expecting a certain property to be available it provides some room to do non-standard things.
> I can always rename the property to _tcti_context instead, to make it clear that it's not expected to by used outside of the package

I would rather not build this on by default. For know folks that need something can just call the underscore variant, we can keep that stable enough. I would like to hear the usecases and fix the API. I look at this as being similar to all the work we did wrapping types.

Reply:
Hm, I have no strong opinion on this. Calling internal-only properties is considered bad practice and shouldn't be done anyway. Currently I cannot think of a use case where a user needs the TCTI cdata. Internal it is, then.